### ISSUE TITLE
isReflectionBasedAssemblyAccessorUsed dosn't work

### ISSUE THREAD

**Issue by devstudiosoft** [Deadpikle/NetSparkle#30]
Hi, i need to use your app without reflection, but seems doesn’t work in my case, i try to use this code in my app

```
Sparkle sparkleUpdateDetector = new Sparkle(user_appcastUrl, icon, SecurityMode.Unsafe, "",
                @"C:\xxxx\xxx.exe")
            {
                TmpDownloadFilePath = Path.Combine(@"C:\xxxx", "Update")
            };
            sparkleUpdateDetector.Configuration = new RegistryConfiguration(@"C:\xxxxxx\xxxx.exe", false);
```

but when i set the isReflectionBasedAssemblyAccessorUsed = false, for use AssemblyDiagnosticsAccessor class, doesn’t work

i updated the source code, and i maked this modification

1) In the constructor of Configuration class first invoke the InitWithDefaultValues() method, and then set the property UseReflectionBasedAssemblyAccessor, otherwise this property remain true.

2) In case of use AssemblyDiagnosticsAccessor (Configuration.cs - row 93) the AssemblyProduct isn't the ApplicationName but the product version, i change the AssemblyDiagnosticsAccessor.cs and the property AssemblyProduct return the variable productName

I attached the two class so, you can compare with the original code and verify if my modification it's right or not, thank you so much

[sparkleModification.zip](https://github.com/Deadpikle/NetSparkle/files/1159267/sparkleModification.zip)

**Comment by Deadpikle** [comment]
Hey @devstudiosoft ! Thanks for the bug report. I will try to take a look at your code in the next few days. Assuming it looks OK, you would be more than welcome to open a pull request to this repository so that you can be marked as a code contributor to the repo and such. If you need more info on how to do that, check out: https://help.github.com/articles/about-pull-requests/

**Comment by Deadpikle** [comment]
Fixed in #31
